src/test/java/ro/secur/auth/security/AuthenticationFilterTest.java

```diff
@@ -0,0 +1,194 @@
+        when(jwtConfigMock.getAuthorizationHeader()).thenReturn(HttpHeaders.AUTHORIZATION);
+        when(jwtConfigMock.getTokenPrefix()).thenReturn("Bearer ");
+    }
+
+    private void mockRequest(JSONObject jsonRequestBody) {
+        requestMock = new MockHttpServletRequest();
+        requestMock.setContentType(MediaType.APPLICATION_JSON);
+        requestMock.setCharacterEncoding("UTF-8");
+        requestMock.setContent(jsonRequestBody.toString().getBytes());
+    }
+
+    private JSONObject createJsonObject() throws JSONException {
+        JSONObject jsonRequestBody = new JSONObject();
+
+        jsonRequestBody.put("username", "test");
+        jsonRequestBody.put("password", "pass");
+
+        return jsonRequestBody;
+    }
+
+    private void mockAuthManagerReturnsAuth() {
+
+        expectedAuth = new UsernamePasswordAuthenticationToken("test", "pass",
+                Collections.singletonList(new SimpleGrantedAuthority(Role.USER.toString())));
+
+        when(authManagerMock.authenticate(any(Authentication.class))).thenReturn(expectedAuth);
+    }
+
+    private void mockAuthManagerReturnsException() {
+
+        when(authManagerMock.authenticate(any(Authentication.class))).thenThrow(AuthenticationException.class);
+    }
+
+    private void mockAuthManagerReturnsBadCredentials() {
+
+        when(authManagerMock.authenticate(any(Authentication.class))).thenThrow(BadCredentialsException.class);
+    }
+
+    @SneakyThrows
+    @Test
+    public void whenAttemptAuthentication_returnAuth() {
+
+        mockAuthManagerReturnsAuth();
+        mockRequest(createJsonObject());
+
+        expectedAuth = new UsernamePasswordAuthenticationToken("test", "pass",
+                Collections.singletonList(new SimpleGrantedAuthority(Role.USER.toString())));
+        Authentication actualAuth = authFilter.attemptAuthentication(requestMock, responseMock);
+
+        assertEquals(expectedAuth, actualAuth);
+    }
+
+//    TODO uncomment this after error handling task is finished
+//    @SneakyThrows
+//    @Test
+//    public void whenAttemptAuthentication_returnException(){
+//
+//        mockAuthManagerReturnsException();
+//        mockRequest(createJsonObject());
+//
+//        assertThrows(AuthenticationException.class, ()->authFilter.attemptAuthentication(requestMock, responseMock));
+//    }
+
+    @SneakyThrows
+    @Test
+    public void whenPerformLogin_returnStatusIsOk() {
+
+        mockAuthManagerReturnsAuth();
+
+        int actualStatus = mockMvc
+                .perform(MockMvcRequestBuilders.post(Api.LOGIN_URL)
+                        .content(createJsonObject().toString().getBytes()))
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse().getStatus();
+
+        assertEquals(HttpStatus.OK.value(), actualStatus);
+    }
+
+    @SneakyThrows
+    @Test
+    public void whenPerformLogin_returnStatusIsUnauthorized() {
```

Comment:
nu ar fi mai ok sa nu pui cu _? .

Reply:
asa am vazut ca se practica. pentru ca trebuie sa pui in numele metodei exact ce testezi si daca nu as folosi _ ar fi mai greu de inteles. asa cumva e separat ce testezi de ce ar trebuie sa se intample